### ISSUE TITLE
Hardcoded cluster domain name

### ISSUE THREAD

**Issue by mp-007** [banzaicloud/kafka-operator#332]
Hi,
cluster domain name is hardcoded as cluster.local
If  your cluster domain name is different, then you cant resolve names of pods

```
func generateKafkaAddress(cluster *v1beta1.KafkaCluster) string {
	if cluster.Spec.HeadlessServiceEnabled {
		return fmt.Sprintf("%s.%s.svc.cluster.local:%d",
			fmt.Sprintf(kafka.HeadlessServiceTemplate, cluster.Name),
			cluster.Namespace,
			cluster.Spec.ListenersConfig.InternalListeners[determineInternalListenerForInnerCom(cluster.Spec.ListenersConfig.InternalListeners)].ContainerPort,
		)
	}
	return fmt.Sprintf("%s.%s.svc.cluster.local:%d",
		fmt.Sprintf(kafka.AllBrokerServiceTemplate, cluster.Name),
		cluster.Namespace,
		cluster.Spec.ListenersConfig.InternalListeners[determineInternalListenerForInnerCom(cluster.Spec.ListenersConfig.InternalListeners)].ContainerPort,
	)
}
```

**Describe the solution you'd like to see**
Adding DomainName parameter in `KafkaClusterSpec struct` and overwrite hardcoded string `cluster.local` in all required files.

https://github.com/banzaicloud/kafka-operator/blob/master/api/v1beta1/kafkacluster_types.go#L33

**Comment by stoader** [comment]
@mp-007 thank you raising this issue. Are you open to take it a stab and create a PR with the proposed change?

**Comment by kaushiksrinivas** [comment]
Hi @stoader 
I would like to work on this enhancement.
Can you please guide me through the process ?

Thanks,
kaushik

**Comment by baluchicken** [comment]
Hi @kaushiksrinivas,
We are glad to hear you are interested in the project.
In the master branch the apis are moved to [pkg/sdk](https://github.com/banzaicloud/kafka-operator/tree/master/pkg/sdk).
The above suggested modification sounds good.
I would make the `DomainName` parameter omitempty to keep backward compatibility.
Please give a default value `cluster.local` to the newly created parameter.
I hope these infos are sufficient to start to work on this.
If not, we are happy to help if you have more questions.
Here is the developer docs if you need help for building the project: https://banzaicloud.com/docs/supertubes/kafka-operator/developer/.
Thanks

**Comment by baluchicken** [comment]
I believe this one is fixes with #361
